### ISSUE TITLE
Unable to upgrade Copilot environment from v1.2.0 to v1.4.0: HTTPS Listener failed to update

### ISSUE THREAD

**Issue by aaroncarlucci** [aws/copilot-cli#2530]
I am managing multiple applications using Copilot and all of them were created using the --domain flag. I've recently upgraded the Copilot version used in my build process, and I'm consistently encountering the same error when Copilot tries to upgrade the environment.
```
✘ Failed to upgrade environment dev's template to version v1.4.0.. 
✘ upgrade environment dev from version v1.2.0 to version v1.4.0: update and wait for stack drone-csup-dev: wait until stack drone-csup-dev update is complete: ResourceNotReady: failed waiting for successful resource state
```

When I look at the CloudFormation stack events, the `HTTPSListener` failed to update with the following message:

```
Resource handler returned message: "Certificate ARN '2021/06/28/[$LATEST]03124a95687c4fd58f341966dbe6d9e2' is not valid (Service: ElasticLoadBalancingV2, Status Code: 400, Request ID: c942874d-8bd3-4faa-b006-d536da4ed9c9, Extended Request ID: null)" (RequestToken: 16666b21-d7a8-b379-77c7-8c3234ff1748, HandlerErrorCode: InvalidRequest)
```

It looks like a CloudWatch path is being substituted for an ACM Certificate ARN for the listener. I briefly looked at the stack and see the HTTPSCert object is a custom Lambda function that presumably outputs this ARN. My guess is there is an issue with that code.

I was able in one instance to rebuild the environment successfully after deleting all the associated services, but I would prefer not to take that route to rebuild three environments each for multiple applications, including prod environments. Any idea with this is happening, or suggestions for a workaround?

---
 EDIT: Furthermore, I looked in the CloudWatch log for what I assume to be the erroneous run of the Lambda function, but there is no useful information there, only:
 ```
2021-06-28T14:45:51.345+02:00 | START RequestId: ec44b656-aaa8-4a7f-b18a-32f63cf7353f Version: $LATEST
-- | --
  | 2021-06-28T14:45:51.515+02:00 | END RequestId: ec44b656-aaa8-4a7f-b18a-32f63cf7353f
  | 2021-06-28T14:45:51.515+02:00 | REPORT RequestId: ec44b656-aaa8-4a7f-b18a-32f63cf7353f Duration: 167.92 ms Billed Duration: 168 ms Memory Size: 512 MB Max Memory Used: 86 MB Init Duration: 457.88 ms
```

**Comment by aaroncarlucci** [comment]
Update: I am able to download Copilot 1.7.1 and run the `copilot env upgrade` command successfully on the legacy environments. So far this seems like a workaround, but I am still convinced there is a bug when trying to hop ahead too many versions without incremental infrastructure upgrades.

I will continue testing my applications and report back and/or close the ticket.

**Comment by iamhopaul123** [comment]
Thank you for reporting the issue. Found the bug and will address it very soon!

**Comment by iamhopaul123** [comment]
Heya we fixed this in our latest release [v1.8.2](https://github.com/aws/copilot-cli/releases/tag/v1.8.2). Thank you for reporting this issue ❤️
